spec/automation/unit/method_validation/validate_quota_spec.rb

```diff
@@ -1,60 +1,130 @@
+    @user = FactoryGirl.create(:user_miq_request_approver)
+    @vm_template = FactoryGirl.create(:template_vmware, :name => "template1")
+    @vm1 = FactoryGirl.create(:vm_vmware)
     @vm1.miq_group = @user.current_group
+    @miq_provision_request = FactoryGirl.create(:miq_provision_request,
+                                                :userid    => @user.userid,
+                                                :src_vm_id => @vm_template.id)
+    @miq_request = @miq_provision_request.create_request
+    @miq_request.save!
   end
 
-  let(:ws) { MiqAeEngine.instantiate("/Infrastructure/VM/Provisioning/StateMachines/ProvisionRequestQuotaVerification/Default?MiqProvisionRequest::miq_provision_request=#{@miq_provision_request.id}&MiqRequest::miq_request=#{@miq_provision_request.id}&max_group_cpu=#{@max_group_cpu}", @user) }
-
   context "validate vcpus quota limit, using number of cpus" do
-    before do
-      prov_options = {:number_of_vms => 1, :owner_email => 'tester@miq.com', :vm_memory => ['1024', '1024'], :number_of_sockets => [2, '2'], :cores_per_socket => [2, '2'], :number_of_cpus => [2, '2']}
-      @miq_provision_request    = FactoryGirl.create(:miq_provision_request, :userid => @user.userid, :src_vm_id => @vm_template.id, :options => prov_options)
-      @miq_request = @miq_provision_request.create_request
-      @miq_request.save!
+    it "no quota limits set" do
+      @quota_limit_max  = YAML.dump(:storage => 0, :vms => 0, :cpu => 0, :memory => 0)
+      @quota_limit_warn = YAML.dump(:storage => 0, :vms => 0, :cpu => 0, :memory => 0)
+      ws = run_automate_method(@miq_provision_request)
+      ws.root['ae_result'].should == 'ok'
     end
+  end
 
-    it "quota failure" do
-      @max_group_cpu = 1
-
-      root = ws.root
-      root['ae_result'].should == 'error'
-      root['ae_state'].should == 'ValidateQuotas'
-      root['reason'].should == "Request denied due to the following quota limits:(Group Allocated vCPUs 0 + Requested 2 > Quota 1) "
+  context "validate max limits" do
+    it "failure max memory" do
+      @quota_limit_max = YAML.dump(:storage => 0, :vms => 0, :cpu => 0, :memory => 4096)
+      @quota_limit_warn = YAML.dump(:storage => 0, :vms => 0, :cpu => 0, :memory => 0)
+      err_msg = "Request denied due to the following quota limits:" \
+                " (memory - Used: 4096 plus requested: 1024 exceeds quota limit: 4096) "
+      ws = run_automate_method(@miq_provision_request)
+      expect(ws.root['ae_result']).to eql('error')
+      @miq_request.reload
+      expect(@miq_request.options[:quota_max_exceeded]).to eql(err_msg)
     end
 
-    it "quota success" do
-      @max_group_cpu = 2
+    it "failure warn memory" do
+      @quota_limit_warn = YAML.dump(:storage => 0, :vms => 0, :cpu => 0, :memory => 4096)
+      @quota_limit_max = YAML.dump(:storage => 0, :vms => 0, :cpu => 0, :memory => 0)
+      err_msg = "Request warning due to the following quota thresholds:" \
```

Comment:
@tinaafitz 
This message has "thresholds" line 67 has "limits". Can we make them consistent